web/src/agama.test.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) [2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import agama from "~/agama";
+
+describe("agama", () => {
+  describe("formatList", () => {
+    afterEach(() => {
+      // restore the default language
+      agama.language = "en";
+    });
+
+    it("it accepts a locale with underscore", () => {
+      agama.language = "zh_Hans";
```

Comment:
This is nonsense. `zh_Hans` with an underscore is neither a POSIX locale nor an IETF language tag

Not sure what the authoritative reference on posix locales is, let's take https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap08.html#tag_08_02

Reply:
Grr, apparently `zh_Hans` comes from Weblate, and we also use that form in the products.d/*.yaml files. I still think it is incorrect.
Note that agama and opensuse-welcome are the only ones using zh_Hans, other projects use zh_CN: https://l10n.opensuse.org/languages/

Reply:
A though from my side: if above comments are right, it's a pity to have hacks in the code instead of aiming for an _upstream_ fix.